### PR TITLE
FIX: EpicsSignalEditMD and state positioner-related fixes

### PR DESCRIPTION
**Pull request by klauer**
## Description

This fixes some issues related to a flurry of recent bug reports. The following changes have been made:
- The MD update signal will be more lenient for cases where specific unset values ("Invalid") persist from the TwinCAT PLC. Devices like TwinCAT state positioners will stop throwing large confusing errors related to invalid enum strings.
- The MD update signal will not send metadata updates until all composite signals are ready and have contributed their values.
- State names are no longer case-sensitive: states like "Out", "out", and "OUT" will be understood to be the same.
- HelpfulIntEnum has been temporarily vendored from pcdsutils/pcdsdaq. This will change to an import in a future release.

## Motivation and Context

- Several state devices have spammy errors when created that interfere with our ability to adjust their enum strings dynamically as viewed from Python.
- Several state devices have variations on the "OUT" string that all mean effectively the same thing, but were previously being interpreted as distinct entities. This lead to some devices being both "IN" and "OUT" because of a misunderstanding of the "Out" string as being "Not "OUT"", for example.
- [Related Jira Ticket](https://jira.slac.stanford.edu/browse/LCLSECSD-550)

## How Has This Been Tested?

The unit tests for pcdsdevices, which covers this section extensively, still pass.
Interactive testing was done to verify that affected real devices now have their behavior corrected, and that related devices have not had their behaviors broken.

## Where Has This Been Documented?

Here in this PR and in release notes. Some docstrings are included where appropriate and necessary.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by tangkong** [comment]
So I don't forget, maybe we should solve case sensitivity issues via `_missing_` or `_missing_value_` methods, similar to how `aenum` does

https://stackoverflow.com/questions/42658609/how-to-construct-a-case-insensitive-enum

**Comment by klauer** [comment]
Vendored helpfulintenum, then pushed it into pcdsutils as well - https://github.com/pcdshub/pcdsutils/pull/44

File: pcdsdevices/state.py [hunk 1]
```diff
@@ -108,7 +108,13 @@ def _state_init(self):
                 self.states_enum = self._create_states_enum()
             self._state_initialized = True
 
-    def _late_state_init(self, *args, enum_strs=None, **kwargs):
+    def _late_state_init(
+        self,
+        *args,
+        connected: bool = False,
```
Comment (ZLLentz) [inline pcdsdevices/state.py:114]: Is this unused now?

File: pcdsdevices/state.py [hunk 2]
```diff
@@ -304,7 +309,7 @@ def _create_states_enum(self):
                 for alias in aliases:
                     state_def[alias] = i
         enum_name = self.__class__.__name__ + 'States'
-        enum = Enum(enum_name, state_def, start=0)
+        enum = HelpfulIntEnum(enum_name, state_def, start=0, module=__name__)
```
Comment (ZLLentz) [inline pcdsdevices/state.py:312]: TIL the enum type constructor has a `module` arg

**Review by ZLLentz — APPROVED** [review]
Small comments, but I think this is actually 100% good if the tests still pass

File: pcdsdevices/state.py (hunk 2 above)
Comment (klauer) [reply pcdsdevices/state.py:312]: Same for me - the ["Warning!"](https://docs.python.org/3.11/howto/enum.html#functional-api) brought it to my attention - unpickleable without it:
image

File: pcdsdevices/state.py (hunk 1 above)
Comment (klauer) [reply pcdsdevices/state.py:114]: Yes, removing it now

File: pcdsdevices/state.py (hunk 2 above)
Comment (ZLLentz) [reply pcdsdevices/state.py:311]: Huh, interesting

**Review by tangkong — APPROVED** [review]
I have no issues with this.  If this clears the angry state error issues then I think our original goal has been met

**Comment by klauer** [comment]
I can't approve my own PR, but the release notes look good @ZLLentz :+1:

**Pull request merged**